Frameworks/CoreGraphics/CGDataProvider.mm

```diff
@@ -156,16 +421,28 @@ void CGDataProviderRelease(CGDataProviderRef data) {
  @Notes
 */
 CGDataProviderRef CGDataProviderRetain(CGDataProviderRef provider) {
-    CFRetain((id)provider);
-
+    CFRetain(provider);
     return provider;
 }
 
 /**
- @Status Stub
+ @Status Interoperable
  @Notes
 */
 CFTypeID CGDataProviderGetTypeID() {
-    UNIMPLEMENTED();
-    return StubReturn();
-}
\ No newline at end of file
+    return __CGDataProvider::GetTypeID();
+}
+
+#pragma endregion // CGDataProvider Functions
+
+#pragma region Internal CGDataProvider Functions
+
+const void* _CGDataProviderGetData(CGDataProviderRef provider) {
+    return provider ? provider->GetData() : nullptr;
+}
+
+size_t _CGDataProviderGetSize(CGDataProviderRef provider) {
+    return provider ? provider->GetSize() : 0;
```

Comment:
it's reasonable but not ideal. we seen issues with merges where ternary operations for conditional checks end up being more work than it needs to be.
usually these conditions do not have tests and regressions come back.

---
In reply to: [93115787](https://github.com/Microsoft/WinObjC/pull/1579#discussion_r93115787) [](ancestors = 93115787,93115443,92931590)